infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/Action.java

```diff
@@ -0,0 +1,66 @@
+package org.corfudb.infrastructure.orchestrator;
+
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.CorfuRuntime;
+
+import javax.annotation.Nonnull;
+import java.util.Map;
+
+/**
+ *
+ * A workflow action. All workflow actions must extend this class.
+ *
+ * Created by Maithem on 10/25/17.
+ */
+
+@Slf4j
+public abstract class Action {
+
+    ActionStatus status = ActionStatus.CREATED;
+
+    /**
+     * Returns the name of this action.
+     * @return Name of action
+     */
+    @Nonnull
+    public abstract String getName();
+
+    /**
+     * The implementation of the action
+     * @param runtime A runtime that the action will use to execute
+     * @throws Exception
+     */
+    public abstract void impl(@Nonnull CorfuRuntime runtime) throws Exception;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)